crates/ra_hir/src/ty/lower.rs

```diff
@@ -298,14 +302,24 @@ fn type_for_enum(db: &impl HirDatabase, s: Enum) -> Ty {
     }
 }
 
+fn type_for_type_alias(db: &impl HirDatabase, t: Type) -> Ty {
+    let generics = t.generic_params(db);
+    let resolver = t.resolver(db);
+    let type_ref = t.type_ref(db);
+    let substs = make_substs(&generics);
+    let inner = Ty::from_hir(db, &resolver, &type_ref);
+    inner.subst(&substs)
+}
+
 #[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
 pub enum TypableDef {
     Function(Function),
     Struct(Struct),
     Enum(Enum),
     EnumVariant(EnumVariant),
+    Type(Type),
```

Comment:
Maybe let's call this `TypeAlias`? `Type` seems ambiguous.

Reply:
Ah, I guess the problem here is me: https://github.com/rust-analyzer/rust-analyzer/commit/0f9c350812e37b3cf2a1fcfb3c60a78122d60bf7#diff-8ce650d0905ec87e4d757e9b1aed9570R348

Should we rename `Type` to `TypeAlias` everywhere?

Reply:
Yeah... Though I'd rather do it as a separate PR :smile: